atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java

```diff
@@ -132,7 +135,7 @@ public String getConnectionPoolName() {
 
     @Value.Default
     public boolean testConnectionBeforeHandout() {
-        return true;
+        return false;
```

Comment:
We have confirmed that disabling resulted in a dramatic performance improvement in our internal authentication service.

Reply:
This seems reasonable

Reply:
A (very slightly) more conservative change would be to enable Hikari's keepAlive feature which pings _idle_ connections in the pool (this is disabled by default - we'd need to provide a ping time).

I don't think that really buys us much - the only time we expect that a connection might go bad is:
- When the DB is actually offline; in that case (assuming there is DB traffic) the connection pool will quickly drain as attempts to use connections will fail, and those connections should meet the condition for eviction based on https://github.com/brettwooldridge/HikariCP/blob/2e07be7b325b05f7225611c60f421e253e1107a5/src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java#L151
- Places that are using an HA setup that don't provide session transfer (this is really the same case, except that we expect that we will be able to create new connections shortly).

I added the testConnectionBeforeHandout config a while back with the expectation that eventually we would see performance problems where this would give us a way out. Now we've seen it - where there is high latency between the client application and DB.